diff --git a/node_modules/@types/chroma-js/index.d.ts b/node_modules/@types/chroma-js/index.d.ts
index c90bb01..43b9da3 100755
--- a/node_modules/@types/chroma-js/index.d.ts
+++ b/node_modules/@types/chroma-js/index.d.ts
@@ -14,13 +14,15 @@ declare namespace chroma {
         hsv: [number, number, number];
         hsi: [number, number, number];
         lab: [number, number, number];
+        oklab: [number, number, number];
         lch: [number, number, number];
         hcl: [number, number, number];
+        oklch: [number, number, number];
         cmyk: [number, number, number, number];
         gl: [number, number, number, number];
     }
 
-    type InterpolationMode = 'rgb' | 'hsl' | 'hsv' | 'hsi' | 'lab' | 'lch' | 'hcl' | 'lrgb';
+    type InterpolationMode = 'rgb' | 'hsl' | 'hsv' | 'hsi' | 'lab' | 'lch' | 'hcl' | 'lrgb' | 'oklab' | 'oklch';
 
     interface ChromaStatic {
         /**
@@ -69,6 +71,11 @@ declare namespace chroma {
 
         lab(lightness: number, a: number, b: number, alpha?: number): Color;
 
+        /**
+         * @see {@link https://bottosson.github.io/posts/oklab/}
+         */
+        oklab(lightness: number, a: number, b: number, alpha?: number): Color;
+
         lch(l: number, c: number, h: number): Color;
 
         /**
@@ -76,6 +83,8 @@ declare namespace chroma {
          */
         hcl(h: number, c: number, l: number): Color;
 
+        oklch(l: number, c: number, h: number): Color;
+
         rgb(r: number, g: number, b: number): Color;
 
         /**
@@ -380,6 +389,16 @@ declare namespace chroma {
          */
         lab: () => ColorSpaces['lab'];
 
+        /**
+         * Returns an array with the **L**, **a**, and **b** components.
+         *
+         * @see {@link https://bottosson.github.io/posts/oklab/}
+         *
+         * @example
+         * chroma('orange').oklab() === [0.79,0.06,0.16]
+         */
+        oklab: () => ColorSpaces['oklab'];
+
         /**
          * Returns an array with the **Lightness**, **chroma**, and **hue**
          * components.
@@ -389,6 +408,7 @@ declare namespace chroma {
          */
         lch: () => ColorSpaces['lch'];
 
+
         /**
          * Alias of [lch](#color-lch), but with the components in reverse
          * order.
@@ -398,6 +418,15 @@ declare namespace chroma {
          */
         hcl: () => ColorSpaces['hcl'];
 
+        /**
+         * Returns an array with the **Lightness**, **chroma**, and **hue**
+         * components.
+         *
+         * @example
+         * chroma('skyblue').lch() === [0.81,0.08,225.75]
+         */
+        oklch: () => ColorSpaces['oklch'];
+
         /**
          * Just like color.rgb but adds the alpha channel to the returned
          * array.
